lib/Common/Memory/Recycler.inl

```diff
@@ -91,7 +91,19 @@ Recycler::AllocWithAttributesInlined(DECLSPEC_GUARD_OVERFLOW size_t size)
     }
 #endif
 
-    char* memBlock = RealAlloc<(ObjectInfoBits)(attributes & InternalObjectInfoBitMask), nothrow>(&autoHeap, allocSize);
+    char* memBlock = nullptr;
+#if GLOBAL_ENABLE_WRITE_BARRIER
+    if (CONFIG_FLAG(ForceSoftwareWriteBarrier)
+        && ((attributes & LeafBit) != LeafBit
+            || (attributes & FinalizeBit) == FinalizeBit)) // there's no Finalize Leaf bucket
```

Comment:
yes, that's why special case this right? :) 
Leaf (without Finalize)->leaf bucket, 
Finalize (whatever leaf or not) -> FinalizeBucket(with barrier here) 

 a little bit confusing though, let me make a more detailed comment in my next change


---
In reply to: [90006103](https://github.com/Microsoft/ChakraCore/pull/2105#discussion_r90006103) [](ancestors = 90006103)